.circleci/config.yml

```diff
@@ -10,8 +10,23 @@ jobs:
       JVM_OPTS: -Xmx3200m
     steps:
       - checkout
+      - run:
+          name: Run Ktlint Checks
+          command: |
+            curl -sSLO https://github.com/pinterest/ktlint/releases/download/0.39.0/ktlint
+            chmod a+x ktlint
+            ./ktlint
       - gradle/with_cache:
           steps:
+            - run:
+                name: Run Detekt Checks
+                command: ./gradlew detektAll
```

Comment:
I'm pretty sure you can run `./gradlew check` and it'll run detekt and tests.

Reply:
Not in this configuration. It would work like this if the `detekt` plugin was applied to all submobules separately instead of only applying it to the top-level `build.gradle`. From personal experience, depending in the project size, running 20 `detekt` tasks is much slower then running the single `detektAll`. That and the overhead of having the plugin code in every `build.gradle` in the project.

It's still possible to make `check` depend on the custom `detektAll` task, of course, if we  want to. I'm not aware of any deep integration between detekt and ktlint (apart from sharing some rules) but running ktlint can be delegated to gradle as well. In which case I can make `check` depend on that task as well.

Reply:
You can add ktlint as a detekt plugin https://detekt.github.io/detekt/#adding-more-rule-sets

Reply:
Thank you, didn't know about this ktlint integration. All folded into `check` now to the best of my groovy-writing abilities.